build.sh

```diff
@@ -112,7 +112,7 @@ prepare_native_build()
     # Generate version.c if specified, else have an empty one.
     __versionSourceFile=$__scriptpath/bin/obj/version.c
     if [ $__generateversionsource == true ]; then
-        $__scriptpath/Tools/corerun $__scriptpath/Tools/MSBuild.exe "$__scriptpath/build.proj" /t:GenerateVersionSourceFile /p:NativeVersionSourceFile=$__scriptpath/bin/obj/version.c /p:GenerateVersionSourceFile=true /v:minimal
```

Comment:
At a high level we want to keep most of the heavy lifting inside our build engine and less in the scripts. We should only pass the minimal amount of arguments to msbuild in order to accomplish the task at hand. 

For this instance `/t:GenerateVersionSourceFile /p:NativeVersionSourceFile=$__scriptpath/bin/obj/version.c /p:GenerateVersionSourceFile=true` there does seem to be a bit of redundancy that doesn't feel like it is needed. It would seem as though we could get away with simply passing `/t:GenerateVersionSourceFile` and the path could be defaulted in the targets file and it doesn't seem like we really need the property if we are explicitly calling this target.

As for figuring out what needs to be passed I think we can still specify that information in the script usage and link to md files for the overall documentation for the dev workflow.